fiftyone/core/odm/database.py

```diff
@@ -1345,11 +1345,26 @@ def _delete_run(dataset_name, run_key, runs_field, run_str, dry_run=False):
 
     if not dry_run:
         _logger.info("Deleting %s doc '%s'", run_str, run_id)
-        conn.runs.delete_one({"_id": run_id})
-        conn.datasets.update_one(
-            {"name": dataset_name},
-            {"$set": {runs_field: runs}},
+        # first remove the run from the dataset
```

Comment:
in case of interruption, delete from the dataset before deleting the run itself to minimize the possibility of referencing runs already deleted